webgui/src/flipmouse/js/tabslot.js

```diff
@@ -35,13 +83,95 @@ tabSlot.initSlots = function () {
     });
 };
 
+tabSlot.initProcesses = function () {
+	if(inElectron === false) {
+		var resultList = ["not available / nicht verfügbar"];
+		//add the process list to the GUI
+		L('.process-select').forEach(function (elem) {
+			elem.innerHTML = L.createSelectItems(resultList);
+		});
+		return;
+	}
+	
+	storage.setDataPath(require('os').homedir());
+	console.log("Looking for config json here: " + storage.getDataPath());
+	
+	//get an active process if one is set for this slot
+	storage.get('FLipMouseGUIConfiguration',function(error, processActivationList) {
+		//if we got a list back (JSON file exists & contains data)
+		if (typeof processActivationList !== 'undefined') {
+			//find active slot in list
+			var index = processActivationList.findIndex(function(item, i){
+				 return item.slot === flip.getCurrentSlot()
+			});
+			activateOnProcess = processActivationList;
+			//if index found -> set name of process
+			if(index !== -1) active = processActivationList[index]["process"];
+			else active = "";
+		} else active = "";
+		
+		ps.lookup({}, function(err, resultList ) {
+			if (err) {
+				throw new Error( err );
+			}
+			//sort list descending by PID (lists recently opened programs first)
+			resultList.sort(function(a, b) {
+				return parseInt(b.pid) - parseInt(a.pid);
+			});
+			//note: on Linux/Mac the first entry is the ps command -> do not list.
+			//resultList.shift();
+			//we just need process names
+			resultList = resultList.map(a => a.command);
+			//if we have an activation process, add to the beginning of the list
+			resultList.unshift(active);
+			//remove duplicates (we trigger on one process name anyway and the list is shorter)
+			resultList = resultList.filter((v, i, a) => a.indexOf(v) === i);
+			//add an "unselected" item at the beginning
+			if(active !== "") resultList.unshift("");
+			
+			//add the process list to the GUI
+			L('.process-select').forEach(function (elem) {
+				elem.innerHTML = L.createSelectItems(resultList);
+				elem.value = active;
+			});
+		});
+	});
+	//TODO: load config into runner...
+};
+
+window.tabSlot.selectProcess = function (select) {
+	if(inElectron === false) return;
+	
+	storage.get('FLipMouseGUIConfiguration',function(error, processActivationList) {
+		var newlist = [{slot: flip.getCurrentSlot(), process: select.value}];
+		if (typeof processActivationList !== 'undefined') {
+				//find active slot in list
+				var index = processActivationList.findIndex(function(item, i){
+					 return item.slot === flip.getCurrentSlot()
+				});
+				//if index found -> update
+				if(index !== -1) processActivationList[index]["process"] = select.value;
```

Comment:
`processActivationList[index].process` is also possible and easier to write/read